packages/mermaid/src/diagrams/flowchart/flowRenderer.js

```diff
@@ -57,10 +58,14 @@ export const addVertices = function (vert, g, svgId, root, _doc, diagObj) {
     if (evaluate(getConfig().flowchart.htmlLabels)) {
       // TODO: addHtmlLabel accepts a labelStyle. Do we possibly have that?
       const node = {
-        label: vertexText.replace(
-          /fa[blrs]?:fa-[\w-]+/g,
-          (s) => `<i class='${s.replace(':', ' ')}'></i>`
-        ),
+        label: vertexText
+          .replace(/fa[blrs]?:fa-[\w-]+/g, (s) => `<i class='${s.replace(':', ' ')}'></i>`)
+          .replace(/\$\$(.*)\$\$/g, (r, c) =>
+            katex
+              .renderToString(c, { throwOnError: true, displayMode: true, output: 'mathml' })
+              .replace(/\n/g, ' ')
+              .replace(/<annotation.*<\/annotation>/g, '')
+          ),
```

Comment:
Let's move this and similar changes to a separate file, so we can apply them to all diagrams.
We would rather not limit fontAwesome or katex support to only flowchart. 

If we use the async loading patter inside that function, we could avoid bundling katex, keeping the size smaller. But that can be a separate PR.

Reply:
Would `diagrams/common/common.ts` be a suitable location?

Reply:
Yes